### PR TITLE
DERP counter

### PR DESCRIPTION
**Pull request by camilovegag**
Closes: https://www.notion.so/hoprnet/DERP-Counter-3e7d5c0def914cfb98a685dc2f0018f8

### Tasks:
- [x] Add counter to DERP table that shows "X calls, mm:ss running"

### Evidence:
image

### CODE REVIEW

File: src/components/DERP/index.jsx [hunk 1]
```diff
@@ -1,33 +1,101 @@
...
+      const seconds = Math.floor(elapsedTime % 60);
+
+      setCounter(
+        `${calls} calls, ${minutes > 9 ? minutes : `0${minutes}`}:${
+          seconds > 9 ? seconds : `0${seconds}`
+        } running`
+      );
+
+    }
+  }, [elapsedTime, currentTime, calls]);
```
Comment (mjadach-iv) [inline src/components/DERP/index.jsx:97]: Why have 2 useEffects for `[elapsedTime, currentTime, calls]` (lines 97 and 66)?
Code looks the same here, has to be fixed.

File: src/sections/Section1.jsx [hunk 2]
```diff
@@ -27,61 +27,72 @@ const Animation = styled.div`
...
+          rel="noopener noreferrer"
+        >
+          RPCh.
+        </a>
+      </Disclaimer>
+    </SSection>
+  );
 }
 
 export default Section1;
```
Comment (mjadach-iv) [inline src/sections/Section1.jsx:98]: Not part of the task, why is it here?

File: src/components/DERP/index.jsx [hunk 3]
```diff
@@ -1,33 +1,101 @@
...
+        startTimeRef.current = Date.now();
+      }
+      setCurrentTime(Date.now());
+      const intervalId = setInterval(() => {
+        setCurrentTime(Date.now());
+      }, 1000);
+
+
+      return () => clearInterval(intervalId);
+    }
```
Comment (mjadach-iv) [inline src/components/DERP/index.jsx:80]: Won't this part of the code will be called every time the `calls` is incremented and bigger than 1? 
The interval constructor should be called only once

File: src/components/DERP/index.jsx [hunk 4]
```diff
@@ -1,33 +1,101 @@
...
+      setElapsedTime((currentTime - startTimeRef.current) / 1000);
+
+      const minutes = Math.floor(elapsedTime / 60);
+      const seconds = Math.floor(elapsedTime % 60);
+
+      setCounter(
+        `${calls} calls, ${minutes > 9 ? minutes : `0${minutes}`}:${
+          seconds > 9 ? seconds : `0${seconds}`
+        } running`
+      );
```
Comment (mjadach-iv) [inline src/components/DERP/index.jsx:94]: Why use it at all and not a react component? Seems like there could be less variables

**Review by mjadach-iv — CHANGES_REQUESTED** [review]

File: src/sections/Section1.jsx (hunk 2 above)
Comment (camilovegag) [reply src/sections/Section1.jsx:98]: that's weird, don't know why this commit went here :/

File: src/components/DERP/index.jsx (hunk 4 above)
Comment (camilovegag) [reply src/components/DERP/index.jsx:94]: ok

**Comment by camilovegag** [comment]
I fixed what you told me @mjadach-iv, thank you

**Review by mjadach-iv — APPROVED** [review]
LGTM

**Pull request merged**